src/librustc/hir/lowering.rs

```diff
@@ -2968,9 +2991,18 @@ impl<'a> LoweringContext<'a> {
                 // for later
                 let ret_def = defs.next().unwrap_or(Def::Err);
 
+                // Here, we are looping over namespaces, if they exist for the definition
+                // being imported. We only handle type and value namespaces because we
+                // won't be dealing with macros in the rest of the compiler.
```

Comment:
> Save-analysis based IDEs should be able to jump ...

They should but we don't support this yet.

> as extra ids in addition to the primary id

When I was debugging, one of the ids was always equal to the primary id (I think that we don't look at the primary id here at all, just these two). I don't understand the stability/deprecation issue, I just added a comment because this seemed pretty non-obvious and took me a while to work out.